TestCombine.playground/Pages/CreatePublisher.xcplaygroundpage/Contents.swift

```diff
@@ -118,3 +118,105 @@ c1?.cancel()
+        guard isCancelled == false else { return .none }
+        return receiveClosure(input)
+    }
+
+    public func receive(completion: Subscribers.Completion<Failure>) -> Void {
+        guard isCancelled == false else { return }
+        return receiveCompletionClosure(completion)
+    }
+
+    fileprivate func cancel() {
+        isCancelled = true
+    }
+}
+
+public struct Orchestrator {
+    public let stop: () -> Void
+    public let start: (Subscribers.Demand) -> Void
+}
+
+public typealias CreationTask<Input, Failure: Error> = (Observer<Input, Failure>, Subscribers.Demand) -> Orchestrator
+
+public struct Create<Output, Failure: Error>: Publisher {
+
+    private let task: CreationTask<Output, Failure>
+    private let demand: Subscribers.Demand
+
+    public init(maxValues: Subscribers.Demand = .unlimited, task: @escaping CreationTask<Output, Failure>) {
+        self.demand = maxValues
+        self.task = task
+    }
+
+    public func receive<S>(subscriber: S) where S : Subscriber, Failure == S.Failure, Output == S.Input {
+        subscriber.receive(subscription: CreateSubscription(publisher: self, subscriber: AnySubscriber(subscriber), maxValues: demand))
+    }
+
+    private class CreateSubscription<Input, Failure: Error>: Subscription {
+
+        @Synchronized private var subscriber: AnySubscriber<Input, Failure>?
+        private var orchestrator: Orchestrator?
+
+        init(publisher: Create<Input, Failure>, subscriber: AnySubscriber<Input, Failure>, maxValues: Subscribers.Demand) {
+            self.subscriber = subscriber
+
+            let observer = Observer<Input, Failure>(
+                receive: { [weak self] input in
+                    self?.subscriber?.receive(input) ?? .none
+                },
+                receiveCompletion: { [weak self] completion in
+                    self?.subscriber?.receive(completion: completion)
+            })
+
+            orchestrator = publisher.task(observer, maxValues)
+        }
+
+        func request(_ demand: Subscribers.Demand) {
+            orchestrator?.start(demand)
+        }
+
+        func cancel() {
+            subscriber = nil
+            orchestrator?.stop()
+        }
+    }
+}
+
+let c = Create<Int, Never>(maxValues: .max(10)) { observer, maxValues in
+
+    var currenctRequest: Subscribers.Demand!
+    let item = DispatchWorkItem {
+
+        let values = [0, 1, 2, 3, 4]
+        var requested = min(currenctRequest, maxValues)
+        for value in values {
+            guard requested > .max(0) else { break }
+            requested -= .max(1)
+            requested = min(requested + observer.receive(value), maxValues)
+        }
+
+        observer.receive(completion: .finished)
+    }
+    
+    return Orchestrator(stop: { item.cancel() },
+                        start: { request in currenctRequest = request; DispatchQueue.global().async(execute: item) })
+}
+.print()
+.sink {
+    print("Value: \($0)")
+}
+//c.cancel()
```

Comment:
Not sure this is a solvable problem in general in multithreaded context.
It's possible that the creation block run on a thread starting pushing values while another thread request the cancel. If you uncomment the cancel you will observe the behaviour some times.
It's possible that the push has been already performed on a thread but the subscription is canceled. As a result we might receive a value after the cancel.
Not sure it's breaking the "rules" since the `cancel` is called as "free up resources" task.

What do you think @drakon-ag ?

Reply:
I don't know if is a best practice to share subscribers among different threads.
But anyway cancel should break the link to the subscriber making subsequents call of receive(Input) NOPs...

Reply:
I found this [post](https://forums.swift.org/t/thread-safety-for-combine-publishers/29491/11) but I have no idea what is the source of that information.
So based on that it should be safe to call that methods from any thread, thus synchronization is required to implement that methods in a thread safe manner.

Following the logic anyway I would assume thread safety as requirement in general. Combine is a framework to unify asynchronous programming and thus means multiple threads by definition.